### ISSUE TITLE
Move game mode into play menu, simplify lap selection

### ISSUE THREAD

**Issue by Wuzzy2** [juzzlin/DustRacing2D#80]
This is an usability-related suggestion.

I think it's a bit weird you have to go into the settings to change the game mode. Also, “changing the lap count” is a bit tedious.

I have a suggestion to simplify this:

* Move game mode from settings to the “Play” menu
* Remove lap selection from “Play” menu
* When selecting “Play”, go through this sequence of screens:
    1. Game mode selection
    2. Difficulty selection
    3. Track selection
* In track selection, use up and down keys and buttons to select number of laps
* Add a little up/down icon next to the lap number to make it clear that up/down keys change the lap count

**Comment by juzzlin** [comment]
I agree otherwise, but wouldn't setting lap count per track change the unlocking logic completely, as now tracks are unlocked by lap count which is always the same for all tracks during a game? Race records etc are also per lap count.

**Comment by Wuzzy2** [comment]
I never said to drop the unlocking mechanic. :)

Basically this up/down buttons are just meant as shortcut. When you change laps, unlock status updates as well.

Or do you mean that would be too confusing?

**Comment by juzzlin** [comment]
Yes, I think it could be a bit confusing if you can change the laps like that. No problems technically to do that :)

**Comment by Korodny** [comment]
How about moving the number of laps decision to the settings menu? Most people probably never change that anyway.

If that's not an option, why not move all of the game modes to the main menu, while combining help and credits into a single menu item:

- one player race
- time trials
- two player race
- two player duell
- settings
- documentation
- quit

or, to keep things more simple, since there might be more game modes in the future:

- one player game
- two player game
- settings
- documentation
- quit

selecting "one" or "two" player game would then present a selection that combines mode + difficulty into one menu:

- race (easy)
- race (medium)
- race (hard)
- time trial (easy)
- ...

which would be followed by laps and track selection, i.e. the number of selections player has to make before a race stays the same.

(time trials are crashing here, I'm assuming that's a single player mode)

**Comment by Korodny** [comment]
Hm, thinking about it some more, IMHO this would be the most ergonomic solution:

- one player game
- two player game
- settings
- documentation
- quit

followed by another menu:

- difficulty: easy/medium/hard
- laps: 3/5/...
- mode: race/time trial
- start racing

followed by the track selection screen.

The game would remember difficulty, number of laps and mode, i.e. save all of them to disk - so I only have to deal with these If I want to change them. In most case, I could just hit ENTER in the 2nd menu, because the default 'cursor' position would be "start racing".

This would probably require some changes to the menu code, so the currently selected option is displayed - that doesn't seem possible right now. But the settings menu would benefit from this as well, IMHO - it's much more user friendly.

**Comment by Wuzzy2** [comment]
> How about moving the number of laps decision to the settings menu? Most people probably never change that anyway.

Nooooo! :-( Number of laps is a very important game setting, don't hide it again.

But combining difficulty, laps and game mode into a single screen sounds like a good idea, as it reduces the number of steps drastically.
